src/ListenerConfigurationChecker.php

```diff
@@ -69,41 +67,17 @@ public function check(array $configuration): void
         }
     }
 
+    /**
+     * @param mixed $definition
+     */
     private function isCallable($definition): bool
     {
-        if (
-            is_array($definition)
-            && array_keys($definition) === [0, 1]
-            && is_string($definition[0])
-        ) {
-            if (class_exists($definition[0])) {
-                try {
-                    $method = new ReflectionMethod($definition[0], $definition[1]);
-                    if ($method->isStatic()) {
-                        return true;
-                    }
-                } catch (ReflectionException $exception) {
-                    return false;
-                }
-            }
-
-            if ($this->container->has($definition[0])) {
-                $object = $this->container->get($definition[0]);
-
-                return method_exists($object, $definition[1]);
-            }
-
+        try {
+            $this->listenerFactory->create($definition);
+        } catch (InvalidListenerConfigurationException $e) {
```

Comment:
It shouldn't be caught as it is caught again and converted to `InvalidListenerConfigurationException` at a higher level (the `check` method). We can just replace `if (!$this->isCallable...` to `$this->listenerFactory->create($definition);`

Reply:
Factory throws two exceptions:
- `InvalidListenerConfigurationException ` on failed create listener, except for internal error in container;
- `ContainerExceptionInterface` on error while retrieving the entry from container.

The `check ` method catches only `ContainerExceptionInterface`, and here we catch `InvalidListenerConfigurationException ` for create more detailed error message.